pystac/summaries.py

```diff
@@ -0,0 +1,170 @@
+
+
+class Summarizer():
+    '''The Summarizer computes summaries from values, following the definition of fields
+    to summarize provided in a json file.
+
+    For more information about the structure of the fields json file, see:
+
+    https://github.com/stac-utils/stac-fields
+
+    Args:
+        fields(str): the path to the json file with field descriptions.
+        If no file is passed, a default one will be used.
+    '''
+    def __init__(self, fields: str = None):
+        if fields is None:
+            self._load_default_field_definitions()
+        else:
+            with open(fields) as f:
+                jsonfields = json.load(f)
+            try:
+                self._set_field_definitions(jsonfields)
+            except:
+                self._load_default_field_definitions()
+
+    def _load_default_field_definitions(self):
+        try:
+            with urllib.request.urlopen(FIELDS_JSON_URL) as url:
+                jsonfields = json.loads(url.read().decode())
+        except:
+            with open(FIELDS_JSON_LOCAL_PATH) as f:
+                jsonfields = json.load(f)
+        self._set_field_definitions(jsonfields)
+
+    def _set_field_definitions(self, fields):
+        self.summaryfields = {}
+        for name, desc in fields["metadata"].items():
+            if isinstance(desc, dict):
+                if desc.get("summary", True):
+                    self.summaryfields[name] = {"mergeArrays": desc.get("mergeArrays", False)}
+            else:
+                self.summaryfields[name] = {"mergeArrays": False}
+
+    def update_with_item(self, summaries, item):
+        for k, v in item.properties.items():
+            if k in self.summaryfields:
+                if isinstance(v, numbers.Number) and not isinstance(v, bool):
+                    rangesummary = summaries.get_range(k, float)
+                    if rangesummary is None:
+                        summaries.add(k, RangeSummary(v, v))
+                    else:
+                        rangesummary.update_with_value(v)
+                elif isinstance(v, list):
+                    listsummary = summaries.get_list(k, Any)
+                    if listsummary is None:
+                        listsummary = []
+                    if self.summaryfields[k]["mergeArrays"]:
+                        listsummary = list(set(listsummary) | set(v))
+                    else:
+                        if v not in listsummary:
+                            listsummary.append(v)
+                    summaries.add(k, listsummary)
+                else:
+                    listsummary = summaries.get_list(k, Any) or []
+                    if v not in listsummary:
+                        listsummary.append(v)
+                    summaries.add(k, listsummary)
+
+
+class Summaries:
+    def __init__(self, summaries: Dict[str, Any], summarizer: Summarizer = None) -> None:
+        self._summaries = summaries
+
+        self.summarizer = summarizer or Summarizer()
```

Comment:
There's currently no caching for the properties json so this would fetch a new document for every collection, which has performance implications.